src/menu.rs

```diff
@@ -178,45 +240,98 @@ pub enum MenuItem {
   /// - **Windows / Android / iOS:** Unsupported
   ///
   Separator,
-}
-
-impl MenuItem {
-  /// Create new custom menu item.
-  /// unique_menu_id is the unique ID for the menu item returned in the EventLoop `Event::MenuEvent(unique_menu_id)`
-  pub fn new(title: impl ToString) -> Self {
-    let title = title.to_string();
-    MenuItem::Custom(CustomMenu {
-      id: MenuId::new(&title),
-      name: title,
-      keyboard_accelerators: None,
-    })
-  }
 
-  /// Assign keyboard shortcut to the menu action. Works only with `MenuItem::Custom`.
+  /// Represents a Submenu(title, enabled, sub menu)
   ///
   /// ## Platform-specific
   ///
-  /// - **Windows / Android / iOS:** Unsupported
+  /// - **Android / iOS:** Unsupported
   ///
+  Submenu(String, bool, MenuPlatform),
```

Comment:
Enum variant can also be a struct. What do you think about change to struct instead?